lib/events/athena/querier.go

```diff
@@ -133,12 +157,33 @@ func (q *querier) SearchSessionEvents(fromUTC, toUTC time.Time, limit int,
 		}
 		filter.condition = condFn
 	}
-	return q.searchEvents(context.TODO(), fromUTC, toUTC, limit, order, startKey, filter, sessionID)
+	events, keyset, err := q.searchEvents(context.TODO(), searchEventsRequest{
+		fromUTC:   fromUTC,
+		toUTC:     toUTC,
+		limit:     limit,
+		order:     order,
+		startKey:  startKey,
+		filter:    filter,
+		sessionID: sessionID,
+	})
+	return events, keyset, trace.Wrap(err)
 }
 
-func (q *querier) searchEvents(ctx context.Context, fromUTC, toUTC time.Time, limit int,
-	order types.EventOrder, startKey string, filter searchEventsFilter, sessionID string,
-) ([]apievents.AuditEvent, string, error) {
+type searchEventsRequest struct {
+	fromUTC, toUTC time.Time
+	limit          int
+	order          types.EventOrder
+	startKey       string
+	filter         searchEventsFilter
+	sessionID      string
+}
+
+func (q *querier) searchEvents(ctx context.Context, req searchEventsRequest) ([]apievents.AuditEvent, string, error) {
+	if q.limiter != nil && !q.limiter.Allow() {
+		// TODO(tobiaszheller): return how much use should sleep at some point.
+		return nil, "", trace.LimitExceeded("rate limit exceeded for searching events")
+	}
```

Comment:
Just want to ensure that we want to rate limit internal searchEvents call instead of eternal Athena API calls.

I'm thinking about the case where `searchEvents` function is invoked for a large sets of `[start, end]` and internally Athena API can be called multiple times by  searchEvents->fetchResults->GetQueryResults within the single `searchEvents` execution.

Reply:
yes, this is fine. Typically the getResults call has much larger quota: https://docs.aws.amazon.com/athena/latest/ug/service-limits.html#service-limits-api-calls
so we should be fine limiting it on searchEvents endpoint.